### ISSUE TITLE
Update Triton pin in PyTorch repo in preparations for 2.9 release

### ISSUE THREAD

**Issue by vlad-penkin** [intel/intel-xpu-backend-for-triton#4912]

**Comment by anmyachev** [comment]
Current status:
* https://github.com/pytorch/pytorch/pull/161050 (**merged**; inductor tests in PyTorch - CI is green)
* **E2E accuracy - good for now**. Still need to check torchbench training test suite: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/17206114420 (CI is in progress).
  * `torchrec_dlrm` - still unable to set up environment for successful launch: https://github.com/intel/intel-xpu-backend-for-triton/pull/4934. It's not a regression, it's happened before. Most likely there won't be any problems with it.
  * `detectron2_fasterrcnn_r_50_fpn` - most likely it's not a problem either, I see a pull request that changes the baseline for this model https://github.com/intel/torch-xpu-ops/pull/1941/files

Results (from https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/17160399700) using https://github.com/intel/intel-xpu-backend-for-triton/pull/4939:
```bash
=========================================
Summary of only failed models:
Real failed models: 1 [['torchrec_dlrm', 'eager_fail_to_run']]
Real failed models: 1 [['torchrec_dlrm', 'eager_fail_to_run']]
Real failed models: 2 [['torchrec_dlrm', 'eager_fail_to_run'], ['detectron2_fasterrcnn_r_50_fpn', 'eager_1st_run_OOM']]
ERROR: Found failed models!
=========================================
ERROR: Missing files detected:
  .//logs-torchbench-float32-training-accuracy/torchbench/float32/inductor_torchbench_float32_training_xpu_accuracy.csv
  .//logs-torchbench-bfloat16-training-accuracy/torchbench/bfloat16/inductor_torchbench_bfloat16_training_xpu_accuracy.csv
  .//logs-torchbench-float16-training-accuracy/torchbench/float16/inductor_torchbench_float16_training_xpu_accuracy.csv
  .//logs-torchbench-amp_bf16-training-accuracy/torchbench/amp_bf16/inductor_torchbench_amp_bf16_training_xpu_accuracy.csv
  .//logs-torchbench-amp_fp16-training-accuracy/torchbench/amp_fp16/inductor_torchbench_amp_fp16_training_xpu_accuracy.csv
Total missing files: 5
```

* **E2E performance suite hasn't launched yet**
* **Windows E2E accuracy hasn't launched yet**
* **E2E huggingface on BMG**: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/17217097168 (to check)

**Comment by etiotto** [comment]
@anmyachev I think you can close this task now.
